### PR TITLE
wmllint: Ignore location_id when checking for shroud and fog

### PR DESCRIPTION
**Pull request by hexagonrecursion**
The following line was causing a wmllint error because it contains _s:
```
Irs, book_start Icr^Ii, Irs
```
```
"../../data/campaigns/Descent_Into_Darkness/maps/07c_A_Small_Favor3.map", line 10: warning, fog or shroud in map file
```

### CODE REVIEW

File: data/tools/wmllint [hunk 1]
```diff
@@ -2884,10 +2884,12 @@ def translator(filename, mapxforms, textxform):
                     else:
                         fields = [x for x in line]
                     outmap.append(fields)
-                    if not maskwarn and maptype == 'map' and re.search('_s|_f(?!me)', line):
-                        print('"%s", line %d: warning, fog or shroud in map file' \
-                              % (filename, lineno+1))
-                        maskwarn = True
+                    terrain_strings = (f.split()[-1] for f in fields)
```
Comment (stevecotton) [inline data/tools/wmllint:2887]: I think this line should be moved inside the outermost conditional

**Comment by stevecotton** [comment]
Thanks for fixing this. BTW, `book_start` is a location_id, not a note.

File: data/tools/wmllint (hunk 1 above)
Comment (hexagonrecursion) [reply data/tools/wmllint:2887]: It took me way too long to figure out which conditional you are referring to as "outermost". I though you meant this one
https://github.com/wesnoth/wesnoth/blob/6a0baa563f8b3a6aede975debd657e6402ec2af5/data/tools/wmllint#L2829
and was trying to understand why or how. But now I see that you were referring to
```py
if not maskwarn and maptype == 'map':
```

Comment (hexagonrecursion) [reply data/tools/wmllint]: Done

**Review by stevecotton — APPROVED** [review]
Looks good to me.

I think this should be merged as soon as the CI build is fixed (a unrelated problem with the vcpkg setup).

**Comment by Pentarctagon** [comment]
Though it hasn't been done the last couple of days, python script changes really should be reviewed by Elvish-Hunter as well.

**Pull request merged**